invenio_rdm_records/services/pids/tasks.py

```diff
@@ -22,3 +24,13 @@ def register_or_update_pid(recid, scheme, parent=False):
         scheme=scheme,
         parent=parent,
     )
+
+
+@shared_task(ignore_result=True)
+def cleanup_parent_pids(recid):
+    """Clean up parent PIDs."""
+    record_cls = current_rdm_records.records_service
+    record = record_cls.pid.resolve(recid)
```

Comment:
I understand that this task is initiated when the `recid` is deleted right? If so, wouldn't that resolve fail?

Reply:
fixed :)